src/Symfony/Component/Console/Style/BufferedOutputStyle.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Style;
+
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Output\TrimmedBufferOutput;
+
+/**
+ * Buffers output to provide additional format helpers based on buffered output.
+ *
+ * @author Vadim Zharkov <hushker@gmail.com>
+ */
+abstract class BufferedOutputStyle extends OutputStyle
```

Comment:
the Symfony style used buffers for implementation reasons.
This is not a behavior, and the name of this class should not be related to it.

Talking about behaviors, I think this could be a `trait` instead of an abstract class

Reply:
Thank you for your review.
What we try to solve with this PR:
1. Make it possible to extend SymfonyStyle and overwrite methods. For this purpose, change some private methods/fields to protected is enough.
2. Decouple some helpers from SymfonyStyle. https://github.com/symfony/symfony/issues/39184#issuecomment-743994682
We could make it with Abstract class or Trait. But extracted helpers couples with OutputStyle (which is Abstract Class), thus it's impossible to implement it as Trait. Further decoupling makes no sense and makes the solution even more overengineered.

Reply:
As per https://github.com/symfony/symfony/issues/39184#issuecomment-734447023 , I propose not to allow extending `SymfonyStyle` (1). The goal here is all about (2): moving the generic helper functions out of `SymfonyStyle` so e.g. `AcmeStyle` can use them as well.

Reply:
Btw, I would love this to be a trait. But I see @hushker's point, especially `newLine()` method overrides makes it quite hard to implement as a trait, right?

Reply:
Reworked without new abstract class.